.gitlab-ci.yml

```diff
@@ -24,39 +24,81 @@ variables:
     - tags
     - schedules
 
-
-.collect_artifacts:                &collect_artifacts 
+.collect_artifacts:                &collect_artifacts
   artifacts:
     name:                          "${CI_JOB_NAME}_${SCHEDULE_TAG:-${CI_COMMIT_REF_NAME}}"
     when:                          on_success
     expire_in:                     1 mos
     paths:
       - artifacts/
 
-test-linux:
+.cargo-check:                      &cargo-check
   stage:                           test
-  variables:
-    RUN_TESTS:                     all
-  script:
-    - scripts/gitlab/test-all.sh
+  before_script:
+    - sccache -s
+  after_script:
     - sccache -s
   tags:
     - linux-docker
 
-test-audit:
+
+cargo-check 0 3:
+  <<:                              *cargo-check
+  script:
+    - time cargo check --target $CARGO_TARGET --locked --no-default-features
+
+cargo-check 1 3:
+  <<:                              *cargo-check
+  script:
+    - time cargo check --target $CARGO_TARGET --locked --manifest-path util/io/Cargo.toml --no-default-features
+
+cargo-check 2 3:
+  <<:                              *cargo-check
+  script:
+    - time cargo check --target $CARGO_TARGET --locked --manifest-path util/io/Cargo.toml --features "mio"
+
+cargo-audit:
   stage:                           test
   script:
-    - set -e
-    - set -u
     - cargo audit
   tags:
     - linux-docker
 
+validate-chainspecs:
+  stage:                           test
+  script:
+    - sccache -s
```

Comment:
could reuse the .cargo_check template (maybe different name)

Reply:
template answer!